### PR TITLE
[27] optimize builtin storage wrappers

### PR DESCRIPTION
**Pull request by icewind1991**
Backport of https://github.com/nextcloud/server/pull/41119

### CODE REVIEW

File: lib/private/Files/FileInfo.php [hunk 1]
```diff
@@ -309,27 +311,12 @@ public function isShareable() {
...
 	 */
 	public function isShared() {
-		$sid = $this->getStorage()->getId();
-		if (!is_null($sid)) {
-			$sid = explode(':', $sid);
-			return ($sid[0] === 'shared');
-		}
-
-		return false;
+		return $this->mount instanceof ISharedMountPoint;
```
Comment (blizzz) [inline lib/private/Files/FileInfo.php:314]: will it not break when files_sharing is disabled?

Comment (juliushaertl) [reply lib/private/Files/FileInfo.php:314]: No, `::class` usage does not require the class to actually exist.

https://www.php.net/manual/en/language.oop5.basic.php#language.oop5.basic.class.class

Comment (blizzz) [reply lib/private/Files/FileInfo.php:314]: it's not `::class` though, I was doubtful about `instanceof`, but it is fine indeed.

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Failure unrelated (drone job timeout)

**Review by Altahrim — APPROVED** [review]

**Pull request merged**